### ISSUE TITLE
Uprade Process : Prevent error on SQL existing row

### ISSUE THREAD

**Issue by camlafit** [PrestaShop/PrestaShop#23433]
#### Describe the bug

Updating from 1.6 in some case, process update broke during SQL insertion, because some date are yet present.

#### Expected behavior

As use case, some modules provide 1.6 and 1.7 compatibility. In this case could enable some hook only present in 1.7 version. If we update from 1.6, we could broke process on different step as at https://github.com/PrestaShop/PrestaShop/blob/6bf6509a2a5a823256c564f1625696d728edd2f3/install-dev/upgrade/sql/1.7.0.0.sql#L147-L155

#### Steps to Reproduce

1. Use a 1.6 versipon
2. Install any plugin with 1.6 and 1.7 comptabillity (and using some new hook as "displayAfterCarrier","displayProductExtraContent","displayNavFullWidth"
2. Use auto upgrade and update to 1.7 stable
3. Run upgrade 
4. Wait broke process

Get an error as : 
```
 SQL 1.7.0.0 1062 in INSERT INTO `ps_hook` (`name`, `title`, `description`, `position`) VALUES ('actionValidateCustomerAddressForm', 'Customer address form validation', 'This hook is called when a customer submit its address form', '1'), ('displayAfterCarrier', 'After carriers list', 'This hook is displayed after the carrier list in Front Office', '1'), ('displayCarrierExtraContent', 'Display additional content for a carrier (e.g pickup points)', 'This hook calls only the module related to the carrier, in order to add options when needed.', '1'), ('validateCustomerFormFields', 'Customer registration form validation', 'This hook is called to a module when it has sent additional fields with additionalCustomerFormFields', '1'), ('displayProductExtraContent', 'Display extra content on the product page', 'This hook expects ProductExtraContent instances, which will be properly displayed by the template on the product page.', '1'), ('displayNavFullWidth', 'Navigation', 'This hook displays full width navigation menu at the top of your pages', '1'), ('displayAfterBodyOpeningTag', 'Very top of pages', 'Use this hook for advertisement or modals you want to load first.', '1'), ('displayBeforeBodyClosingTag', 'Very bottom of pages', 'Use this hook for your modals or any content you want to load at the very end.', '1'): Duplicate entry 'displayAfterCarrier' for key 'hook_name'
```

#### Solution proposed

Use SQL IGNORE keyword on any SQL insertion

#### Additional information

* PrestaShop version: 1.6.1.x
* PHP version: 7.1

**Comment by camlafit** [comment]
Note : partially done with bdb65673dcafc0fcd8899d137431a62d165487bf

**Comment by florine2623** [comment]
Hello @camlafit ,

On step 2, are you talking about PS native module or external module ?
Thanks

**Comment by camlafit** [comment]
hello @florine2623 

In our use case, It's related to an external module

**Comment by hibatallahAouadni** [comment]
Hello,

PrestaShop 1.6 has reached end of life and is no longer officially maintained ([see announcement](http://build.prestashop.com/news/1.6.1.x-what-s-next/)).

Maintenance for this version is now being performed exclusively by volunteers on a very limited scope. You can open an issue on its [dedicated repository](https://github.com/PrestaShop/PrestaShop-1.6/issues), but please keep in mind that only absolutely critical issues may be acknowledged.

Thank you

**Comment by camlafit** [comment]
Hello

We talk about process to upgrade to 1.7 not to maintain 1.6. I don't understand why this issue should be close. 
All other SQL queries about ps_hooks use IGNORE statement, this PR respect existing SQL 

This PR as any impact and allow user to use maintained version on some use case.

**Comment by hibatallahAouadni** [comment]
Hello @camlafit 

It's recommended to unisntall all the modules hose not compatible with 1.7 before updating.

> Install any plugin with 1.6 and 1.7 comptabillity

It's not clear if you're talking about the modules that are compatible with both version?
Could you please provide an exemple?

Thanks!

**Comment by camlafit** [comment]
Hello

When a module is uninstalled, all hooks declared in database remain. Then uninstall before any module doesn't change ps_hooks table content. In our test we have set autoupgrade to disable any non standard module.

I do a grep on our server and found some plugins using at least once hook in this list displayBeforeBodyClosingTag|displayAfterBodyOpeningTag|validateCustomerFormFields|displayCarrierExtraContent|actionValidateCustomerAddressForm|displayAfterCarrier|displayProductExtraContent|displayNavFullWidth 

modules/chronopost/upgrade/Upgrade-4.5.0.php
modules/ets_megamenu/upgrade/install-2.0.5.php
modules/homecomments/upgrade/install-3.0.0.php
modules/productvideo/upgrade/upgrade-2.7.9.php

These modules are took from official addons marketplace.

**Comment by camlafit** [comment]
I detect also a crash on update process on same file with 

```
/* PHP:delete_hook(displayProductTab); */;
/* PHP:delete_hook(displayProductTabContent); */;
/* PHP:delete_hook(displayBeforePayment); */;
/* PHP:delete_hook(actionBeforeAuthentication); */;
/* PHP:delete_hook(actionOrderDetail); */;
/* PHP:delete_hook(actionProductListOverride); */;
/* PHP:delete_hook(actionSearch); */;
/* PHP:delete_hook(displayCustomerIdentityForm); */;
/* PHP:delete_hook(displayHomeTab); */;
/* PHP:delete_hook(displayHomeTabContent); */;
/* PHP:delete_hook(displayPayment); */;
/* PHP:delete_hook(displayProductComparison); */;
```

But any notice on debug
In PR I propose a change with full SQL DELETE as method is use only once on all code

**Comment by hibatallahAouadni** [comment]
Hello @camlafit 

After discussing your issue with @atomiix it seems valid.
I will reopen the issue :wink: 

Ping @PrestaShop/prestashop-core-developers could you please take a look :wink: thanks in advance

Thanks!

**Comment by khouloudbelguith** [comment]
Hi @camlafit, @atomiix,

I did not manage to reproduce the issue after an upgrade from PS16124 to PS1772 using the 1-click upgrade v4.12.0.
Steps that I followed:
1. Install 16124
2. Install a module using those hooks
```
        $this->registerHook('displayProductTab');
        $this->registerHook('displayPayment');
        $this->registerHook('displayHomeTab');
        $this->registerHook('displayBeforePayment');
```
3. I installed the Product video module also
4. Do an upgrade to PS177 => ok
5. I checked the first module, it is not hooked to those hooks anymore

I attached a screen record
https://drive.google.com/file/d/1Fi1TDcgSGgQunAF_hh30Uvoa9fhApYxU/view?usp=sharing

I used php7.1.
Here's my log
[log.txt](https://github.com/PrestaShop/PrestaShop/files/6183865/log.txt)


Thank you!
